### ISSUE TITLE
Help with understanding error: NotImplementedError

### ISSUE THREAD

**Issue by engineertree5** [matplotlib/mplfinance#375]
OS: Raspberry pi3
System: 

```
NAME="Raspbian GNU/Linux"
VERSION_ID="10"
VERSION="10 (buster)"
VERSION_CODENAME=buster
ID=raspbian
ID_LIKE=debian
```

Running mplfinance in headless-mode on my raspberry pi and I'm trying to wrap my head around this error (below) i'm getting. The bulk of the error states `NotImplementedError: kwarg "style" is NOT YET implemented.` which I'm not sure how to break down. I read the pydocs [here](https://docs.python.org/3/library/exceptions.html#NotImplementedError) but still a bit lost where the error is.

> In user-defined base classes, abstract methods should raise this exception when they require derived classes to override the method, or while the class is being developed to indicate that the real implementation still needs to be added.

I could modify the module but that is a can of worms I'm trying to avoid. I'm semi-new to python and trying to work my way through this codebase + docs and posting this for help/awareness. Let me know if you need more info to help with this. 

> FYI: Running this on my Mac works without error

FULL error below:
```bash
  File "/home/pi/.local/lib/python3.7/site-packages/mplfinance/plotting.py", line 181, in plot
    config = _process_kwargs(kwargs)
  File "/home/pi/.local/lib/python3.7/site-packages/mplfinance/plotting.py", line 160, in _process_kwargs
    raise NotImplementedError('kwarg "'+key+'" is NOT YET implemented.') 
NotImplementedError: kwarg "style" is NOT YET implemented.
```

**Comment by DanielGoldfarb** [comment]
Please do
```
import mplfinance as mpf
print(mpf.__version__)
```

Please report the result here.

**Comment by engineertree5** [comment]
Daniel,

Thanks for responding! I'm printing my results below 

```python
>>> import mplfinance as mpf
Unable to init server: Could not connect: Connection refused
Unable to init server: Could not connect: Connection refused

(.:2381): Gdk-CRITICAL **: 21:05:07.513: gdk_cursor_new_for_display: assertion 'GDK_IS_DISPLAY (display)' failed

(.:2381): Gdk-CRITICAL **: 21:05:07.519: gdk_cursor_new_for_display: assertion 'GDK_IS_DISPLAY (display)' failed
```

Just a reminder I'm running headless on my raspberry pi3

```python
print(mpf.__version__)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: module 'mplfinance' has no attribute '__version__'
```

I'm not seeing `__version__` as an option here

```python
>>> dir(mpf)
['__builtins__', '__cached__', '__doc__', '__file__', '__loader__', '__name__', '__package__', '__path__', '__spec__', '_utils', 'plot', 'plotting']
```

exited repel and did a `pip3 list` == `mplfinance==0.11.0`

I'm certain it has to do with me running this on my pi. i'll look into the GDK error. Any idea what the `Unable to init server: Could not connect: Connection refused` is about?

**Comment by DanielGoldfarb** [comment]
It's tricky to run any package that uses matplotlib in a headless installation, but it can be done.  In short, a headless environment has no graphics, but plotting candlesticks ***is*** graphics.   It can be done by using matplotlib backend "Agg", and saving the plot to an image file.  You may have to `export MPLBACKEND=Agg` [as noted here](https://github.com/matplotlib/mplfinance/issues/331#issuecomment-776173521)

But **it also appears that you have a *very* old version of mplfinance**.  Please do:
```
pip3 install --upgrade mplfinance
```
(you *may* have to `pip3 uninstall mplfinance` first)

Then, to understand how you can run mplfinance (or matplotlib in general) in a headless environment, [**see Issue 331**](https://github.com/matplotlib/mplfinance/issues/331), particularly [**this comment**](https://github.com/matplotlib/mplfinance/issues/331#issuecomment-775995246).

HTH.  --Daniel

**Comment by engineertree5** [comment]
EXCITED.gif

We are operational, Daniel! here is what I did
1. upgraded mplfinance like you said. I had to give it the `==`. some reason the pip3 on the pi was being stubborn. 

```bash
pip3 install --upgrade mplfinance
```

2. exported env var `MPBLACKEND`

```bash
export MPLBACKEND=Agg
```

3. tell matplotlib to use env var

```python
matplotlib.use('Agg')
```

after that, I was able to save the image as .png and tweet it out! Watchout Skynet

Thank you for taking the time to help me this late Sunday evening (EST). Really saved a lot of troubleshooting + educated me on knowing the pitfalls of running headless.
